server/services/property.service.js

```diff
@@ -265,3 +268,40 @@ export const getAvailablePropertyOptions = async () =>
 
 export const getPropertiesWithPaymentPlanId = async (id) =>
   Property.find({ paymentPlan: { $in: [id] } });
+
+export const getTotalAmountPaidForProperty = async (offerId) =>
+  Transaction.aggregate([
+    { $match: { offerId: ObjectId(offerId) } },
+    {
+      $lookup: {
+        from: 'properties',
+        localField: 'propertyId',
+        foreignField: '_id',
+        as: 'property',
+      },
+    },
+    {
+      $unwind: '$property',
+    },
+    {
+      $group: {
+        _id: '$property._id',
+        totalAmountContributed: { $sum: '$amount' },
+      },
+    },
+  ]);
+
+export const getAssignedPropertyById = async (offerId) => {
+  const total = await getTotalAmountPaidForProperty(offerId);
+  const totalPaid = total.length > 0 ? total[0].totalAmountContributed : 0;
+
+  const offer = await getOfferById(offerId);
+
+  const property = await getPropertyById(offer.propertyId);
+
+  return {
+    totalAmountPayable: offer.totalAmountPayable,
```

Comment:
Looking at this, I think it is more preferable to return the offer to cater for future UI changes instead of the `totalAmountPayable` only. For example, offerId is needed in the frontend to make the links work appropriately but it is missing.

```suggestion
    offer,
```